gateway/src/main/java/com/lyft/data/gateway/router/QueryHistoryManagerImpl.java

```diff
@@ -0,0 +1,31 @@
+package com.lyft.data.gateway.router;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class QueryHistoryManagerImpl implements QueryHistoryManager {
+
+  private int size = 2000; // TODO: take this input from config.
```

Comment:
For context for anyone else looking at the temporary magic number: looking at grafana i'm guessing this'll retain ~a day of query history information — that's probably good enough for answering "what cluster did my query go to" styles of questions.